exe.c

```diff
@@ -101,6 +117,13 @@ void jump_rel(int offset);
 void call(int lbl);
 void ret();
 
+void dup(int reg) {
+  pop_reg(reg);
```

Comment:
How does the register system work in pnut? I haven't looked at it yet. Is the `int` just a mapping to an `enum`?

Reply:
The exe backend uses a stack machine so we don't need to do register allocation. Still, the exe backend declares 3 general purpose registers, `regX`, `regY` and `regZ`, [represented using ints](https://github.com/udem-dlteam/pnut/blob/master/exe.c#L87-L91) and [defined in the x86 file](https://github.com/udem-dlteam/pnut/blob/master/x86.c#L21-L46).